awscli/customizations/paginate.py

```diff
@@ -164,23 +164,32 @@ def unify_paging_params(argument_table, operation_model, event_name,
                                      parse_type=type_name,
                                      serialized_name='MaxItems'),
                         shadowed_args)
-    # We will register two pagination handlers.
+
+    # We will register three pagination handlers.
     #
     # The first is focused on analyzing the CLI arguments passed to see
     # if they contain explicit operation-specific pagination args.  If so,
     # they are doing manual pagination and we turn off the CLI pagination.
     #
-    # The second is called later in the event chain and analyzes the actual
+    # The second analyzes the CLI arguments to see if the user has specified
+    # a valid value for the max-items argument, and warns them if not.
+    #
+    # The third is called later in the event chain and analyzes the actual
     # calling parameters passed to the operation.
     #
-    # The reason we have to do the second is that someone could put
+    # The reason we have to do the third is that someone could put
     # operation-specific pagination args in the CLI input JSON file
     # directly and this bypasses all of the CLI args processing.
     session.register(
         parsed_args_event,
         partial(check_should_enable_pagination,
                 list(_get_all_cli_input_tokens(paginator_config)),
                 shadowed_args, argument_table))
+
+    session.register(
```

Comment:
I'm not sure it makes sense to do the validation here. Mainly, this function is responsible for ensuring the AWS CLI's custom paging parameters aren't conflicting with the underlying service model parameters.

Instead, let's defer the validation until the max items object for `PageArgument` calls `add_to_params`. This also makes it more explicit that the parameter we're validating is for the paginator, and not some parameter named `MaxItems` that a service model defined.

Reply:
Thanks, I addressed this in my latest revision.